compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/partitionning.rs

```diff
@@ -333,6 +368,269 @@ pub mod tests {
         }
     }
 
+    fn get_external_partition_index_or(
+        external_partition: &ExternalPartition,
+        p_cut: &PartitionCut,
+        default: usize,
+    ) -> PartitionIndex {
+        for (i, e_partition) in p_cut.external_partitions.iter().enumerate() {
+            if external_partition == e_partition {
+                return PartitionIndex(i + p_cut.n_internal_partitions());
+            }
+        }
+        PartitionIndex(default)
+    }
+
+    pub const GLWE_PARAMS: GlweParameters = GlweParameters {
+        log2_polynomial_size: 11,
+        glwe_dimension: 1,
+    };
+
+    pub const TFHERS_PBS_LEVEL: u64 = 1;
+
+    pub const TFHERS_MACRO_PARAMS: MacroParameters = MacroParameters {
+        glwe_params: GLWE_PARAMS,
+        internal_dim: 841,
+    };
+
+    pub fn get_tfhers_max_noise() -> f64 {
+        get_noise_br(
+            SHARED_CACHES.caches(),
+            GLWE_PARAMS.log2_polynomial_size,
+            GLWE_PARAMS.glwe_dimension,
+            TFHERS_MACRO_PARAMS.internal_dim,
+            TFHERS_PBS_LEVEL,
+            None,
+        )
+        .unwrap()
+    }
+
+    #[test]
+    fn test_tfhers_in_out_dot_compute() {
+        let mut dag = unparametrized::Dag::new();
+        let input1 = dag.add_input(16, Shape::number());
+        let variance = get_tfhers_max_noise();
+        let tfhers_partition = ExternalPartition {
+            name: String::from("tfhers"),
+            macro_params: TFHERS_MACRO_PARAMS,
+            max_variance: variance * 4.0,
+            variance,
+        };
+        let change_part1 = dag.add_change_partition(input1, Some(tfhers_partition.clone()), None);
+        let dot = dag.add_dot([change_part1], [2]);
+        _ = dag.add_change_partition(dot, None, Some(tfhers_partition));
+
+        let partitions = partitionning(&dag);
+        assert!(partitions.nb_partitions == 1);
+        let instrs_partition = partitions.instrs_partition;
+        show_partitionning(&dag, &instrs_partition);
+        assert!(instrs_partition[0].instruction_partition == LOW_PRECISION_PARTITION);
```

Comment:
Maybe add an assert to show that the exterenal partition is used and no internal partition exists.